File: reversed_binary_numbers/python/reversed_binary_numbers.py
#!/usr/local/bin/python

class ReversedBinaryNumbers:
  @staticmethod
  def convert_decimal_to_binary(decimal):
    return int(bin(decimal)[2:])

  @staticmethod
  def reverse_binary(binary):
    return int(str(binary)[::-1])

  @staticmethod
  def convert_binary_to_decimal(binary):
    return int(str(binary), 2)

  @staticmethod
  def reverse(decimal):
    return ReversedBinaryNumbers.convert_binary_to_decimal(ReversedBinaryNumbers.reverse_binary(ReversedBinaryNumbers.convert_decimal_to_binary(decimal)))

def main():
  number = input()
  print(ReversedBinaryNumbers.reverse(number))

main()

File: reversed_binary_numbers/ruby/reversed_binary_numbers.rb
class ReversedBinaryNumbers
  def self.reverse(decimal)
    convert_binary_to_decimal(reverse_binary(convert_decimal_to_binary(decimal)))
  end

  def self.convert_decimal_to_binary(decimal)
    decimal.to_s(2).to_i
  end

  def self.convert_binary_to_decimal(binary)
    binary.to_s.to_i(base=2)
  end

  def self.reverse_binary(binary_number)
    binary_number.to_s.reverse.to_i
  end
end